pkg/components/internal/testutil/jsonpath.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	yamlserializer "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/util/jsonpath"
+)
+
+// getUnstructredObj inputs any Kubernetes manifest in YAML format and returns an object of type
+// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
+// extract metadata of the Kubernetes manifest.
+func getUnstructredObj(t *testing.T, yamlObj string) *unstructured.Unstructured {
+	u := &unstructured.Unstructured{}
+
+	// Decode YAML into `unstructured.Unstructured`.
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode([]byte(yamlObj), nil, u); err != nil {
+		t.Fatalf("Converting config to unstructured.Unstructured: %v", err)
+	}
+
+	return u
+}
+
+// getValFromObject takes JSON path as string and an object of type unstructured.Unstructured. This
+// function returns object of type reflect.Value at that JSON path.
+func getValFromObject(t *testing.T, jp string, obj *unstructured.Unstructured) reflect.Value {
+	jPath := jsonpath.New("parse")
+	if err := jPath.Parse(jp); err != nil {
+		t.Fatalf("Parsing JSONPath: %v", err)
+	}
+
+	v, err := jPath.FindResults(obj.Object)
+	if err != nil {
+		t.Fatalf("Finding results using JSONPath in the YAML file: %v", err)
+	}
+
+	if len(v) == 0 || len(v[0]) == 0 {
+		t.Fatalf("No result found")
+	}
+
+	return v[0][0]
+}
+
+// getJSONPathValue extracts an object at a JSON path from a YAML config, and returns interface
+// object.
+func getJSONPathValue(t *testing.T, yamlConfig string, jsonPath string) interface{} {
+	u := getUnstructredObj(t, yamlConfig)
+	got := getValFromObject(t, jsonPath, u)
+
+	switch got.Kind() { //nolint:exhaustive
+	case reflect.Interface:
+		// TODO: Add type switch here for concrete types.
+		return got.Interface()
+	default:
+		t.Fatalf("Unknown type of the object extracted from the converted YAML: %v", got.Kind())
+	}
+
+	return nil
+}
+
+// MatchJSONPathStringValue is a helper function for component's unit tests. It compares a string
+// value of a JSON path in a YAML config.
```

Comment:
```suggestion
// MatchJSONPathStringValue is a helper function for component unit tests. It compares a string
// value of a JSON path in a YAML config.
```

>MatchJSONPathStringValue is a helper function for component's unit tests.

Aren't all the functions in this file helpers for component unit tests?

>It compares a string value of a JSON path in a YAML config

Can we rephrase this? I don't understand what this means.

Reply:
> Aren't all the functions in this file helpers for component unit tests?

No, other functions in this file aren't exported.

> Can we rephrase this? I don't understand what this means.

Sure.

Reply:
Thanks. I still don't understand the last part of the comment but let's not block on this.